File: php/150-evaluate-reverse-polish-notation.php
<?php
class Solution 
{
    /**
     * @param String[] $tokens
     * @return Integer
     */
    public function evalRPN($tokens) 
    {
        $stack = [];
        foreach ($tokens as $token) {
            if (is_numeric($token)) {
                array_push($stack, $token);
            } else {
                $num1 = array_pop($stack);
                $num2 = array_pop($stack);
                switch ($token) {
                    case '+':
                        array_push($stack, $num1+$num2);
                        break;
                    case '-':
                        array_push($stack, $num2-$num1);
                        break;
                    case '*':
                        array_push($stack, $num2*$num1);
                        break;
                    case '/':
                        array_push($stack, intval($num2/$num1));
                        break;
                    default:
                        throw new \Exception("wrong token.");
                }
            }
        }
        return array_pop($stack);
    }
}

File: go/150-evaluate-reverse-polish-notation.go
package main

func evalRPN(tokens []string) int {
	if len(tokens) == 0 {
		return 0
	}
	var stack []int
	for _, t := range tokens {
		switch t {
		case "+":
			v1 := int(stack[len(stack)-1])
			v2 := int(stack[len(stack)-2])
			res := int(v1) + int(v2)
			stack = stack[:len(stack)-2]
			stack = append(stack, res)
		case "-":
			v1 := int(stack[len(stack)-1])
			v2 := int(stack[len(stack)-2])
			res := v2 - v1
			stack = stack[:len(stack)-2]
			stack = append(stack, res)
		case "*":
			v1 := int(stack[len(stack)-1])
			v2 := int(stack[len(stack)-2])
			res := v1 * v2
			stack = stack[:len(stack)-2]
			stack = append(stack, res)
		case "/":
			v1 := int(stack[len(stack)-1])
			v2 := int(stack[len(stack)-2])
			res := v2 / v1
			stack = stack[:len(stack)-2]
			stack = append(stack, res)
		default:
			number, _ := strconv.Atoi(t)
			stack = append(stack, number)
		}
	}
	return stack[0]
}
